File: LeetCode Solutions/February 2021 LeetCoding Challenge/Score of Parentheses/Score of Parentheses.cpp
class Solution {
public:
    int scoreOfParentheses(string S) {
        int power = 0, score = 0;
        for (int i=1; i<S.length(); i++){
            if (S[i] == '('){
                power++;
            }
            else if (S[i-1] == '('){
                score += 1 << power--;
            } else{
                power--;
            }
        }
        return score;
    }
};

File: LeetCode Solutions/March 2022 LeetCoding Challenge/Score of Parentheses/Score of Parentheses.py
class Solution:
    def scoreOfParentheses(self, s: str) -> int:
        power, score = 0, 0
        for i in range(1, len(s)):
            if (s[i] == "("):
                power += 1
            elif (s[i-1] == "("):
                score += 1 << power
                power -= 1
            else:
                power -= 1
        return score